libumccr/aws/libsm.py

```diff
@@ -6,12 +6,24 @@
 import base64
 import logging
 
-from cachetools.func import lru_cache
-
 from libumccr.aws import sm_client
+from libumccr.utils import load_package_if_found
 
 logger = logging.getLogger(__name__)
 
+if load_package_if_found("cachetools"):
+    logger.info(f"cachetools found, using LRU cache")
+    from cachetools.func import lru_cache
+else:
+    def lru_cache(maxsize):
+        logger.info(f"cachetools not found, skipping LRU cache with maxsize={maxsize}")
+
+        def wrapper(func):
+            func.cache_clear = lambda: None
+            return func
+
+        return wrapper
```

Comment:
Higher order function and function object attribute trick here, Flo. 
See if you could traceback this puzzle piece. @reisingerf 😉